### ISSUE TITLE
nan values during training

### ISSUE THREAD

**Issue by mfiore** [zhreshold/mxnet-ssd#147]
Hello! 
Congratulation for this very cool implementation of ssd! I have tried running the demo and training the model on pascal voc and it worked perfectly. I was now trying to train my own dataset and had some problems.

I have a dataset in the pascal voc format, which contains 2 classes. I have already trained it on ssd_caffe and it works fine. To train it with mxnet I created my own dataset class, inheriting from imdb, changed the prepare_dataset.py script to include this class, and generated my own .rec files. I then used the train.py function, setting --num-class to 2, --class-names to my class names, and --num-example to the number of examples in my dataset. 

Both the CrossEntropy and the SmoothL1 go to nan at the second step. I tried changing the learning rate to very low values but I still have this problem. I also tried removing all the other learning parameters (momentum etc.) but still no luck.

Any idea of what could be the problem? My guesses are two:
- my conversion to imdb has some problem. Any way to check this?
- i used as pre-trained model vgg16_reduced which has 20 classes, while I'm only using two (but I do the same on caffe-ssd and it works there)

Thansk a lot for any help and for this tool!

**Comment by zhreshold** [comment]
Since you have tried tuning learning rate without luck, then probably there's a problem with the custom dataset implementation. I would suggest print out the loaded labels, and see if they are reasonable, i.e., label is a matrix with format [[id, xmin, ymin, xmax, ymax], [id, ...], ...].
Note that the xmin is in range [0, 1]

**Comment by mfiore** [comment]
Thank you very much! Looks like the error was in the label format. I was not normalizing the bounding box coordinates.
